File: python_mid_june_2024/122. Best Time to Buy and Sell Stock II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        buy_at = prices[0]

        for price in prices:
            if price < buy_at:
                buy_at = price
            else:
                max_profit += price - buy_at
                buy_at = price
        return max_profit

File: java_july/122. Best Time to Buy and Sell Stock II.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyAt = Integer.MAX_VALUE;
        int res = 0;

        for (int price : prices) {
            if (price < buyAt) {
                buyAt = price;
            } else {
                res += price - buyAt;
                buyAt = price;
            }
        }
        return res;
    }
}